### PR TITLE
Retrieve schema table names

### PR DESCRIPTION
**Pull request by dswiecki**

### CODE REVIEW

File: engine/components/sql/src/main/scala/pl/touk/nussknacker/sql/db/schema/DbMetaDataProvider.scala [hunk 1]
```diff
@@ -23,10 +25,28 @@ class DbMetaDataProvider(getConnection: () => Connection) {
       } finally statement.close()
     } finally conn.close()
   }
+
+  def getSchemaDefinition(): SchemaDefinition = {
+    val connection = getConnection()
```
Comment (mproch) [inline engine/components/sql/src/main/scala/pl/touk/nussknacker/sql/db/schema/DbMetaDataProvider.scala:30]: where do we close this connection? :)

File: engine/components/sql/src/main/scala/pl/touk/nussknacker/sql/db/schema/DbMetaDataProvider.scala [hunk 2]
```diff
@@ -23,10 +25,28 @@ class DbMetaDataProvider(getConnection: () => Connection) {
...
+    val tables = connection.getMetaData.getTables(null, "PUBLIC", "%", Array("TABLE").map(_.toString))
```
Comment (mproch) [inline engine/components/sql/src/main/scala/pl/touk/nussknacker/sql/db/schema/DbMetaDataProvider.scala:31]: Is it DB agnostic? I'm afraid it's not... schema pattern etc. also should probably be configurable?

File: engine/components/sql/src/main/scala/pl/touk/nussknacker/sql/db/schema/DbMetaDataProvider.scala [hunk 3]
```diff
@@ -23,10 +25,28 @@ class DbMetaDataProvider(getConnection: () => Connection) {
...
+    val results = new util.ArrayList[String]()
+    val columnNameIndex = 3
+    while (tables.next()) {
```
Comment (mproch) [inline engine/components/sql/src/main/scala/pl/touk/nussknacker/sql/db/schema/DbMetaDataProvider.scala:34]: maybe some scala conversions would be easier?

File: engine/components/sql/src/main/scala/pl/touk/nussknacker/sql/db/schema/DbMetaDataProvider.scala [hunk 4]
```diff
@@ -23,10 +25,28 @@ class DbMetaDataProvider(getConnection: () => Connection) {
...
+      val str: String = tables.getString(columnNameIndex)
+      results add str
+    }
+    SchemaDefinition(results)
+  }
 }
 
 case class DialectMetaData(identifierQuote: String)
 
+case class SchemaDefinition(tables: java.util.List[String])
```
Comment (mproch) [inline engine/components/sql/src/main/scala/pl/touk/nussknacker/sql/db/schema/DbMetaDataProvider.scala:44]: Do we need java list here?

File: engine/components/sql/src/main/scala/pl/touk/nussknacker/sql/db/schema/DbMetaDataProvider.scala (hunk 1 above)
Comment (dswiecki) [reply engine/components/sql/src/main/scala/pl/touk/nussknacker/sql/db/schema/DbMetaDataProvider.scala:30]: Done

File: engine/components/sql/src/main/scala/pl/touk/nussknacker/sql/db/schema/DbMetaDataProvider.scala (hunk 3 above)
Comment (dswiecki) [reply engine/components/sql/src/main/scala/pl/touk/nussknacker/sql/db/schema/DbMetaDataProvider.scala]: Done

File: engine/components/sql/src/main/scala/pl/touk/nussknacker/sql/db/schema/DbMetaDataProvider.scala (hunk 4 above)
Comment (dswiecki) [reply engine/components/sql/src/main/scala/pl/touk/nussknacker/sql/db/schema/DbMetaDataProvider.scala]: DOne

File: engine/components/sql/src/main/scala/pl/touk/nussknacker/sql/db/schema/DbMetaDataProvider.scala [hunk 5]
```diff
@@ -23,10 +23,41 @@ class DbMetaDataProvider(getConnection: () => Connection) {
       } finally statement.close()
     } finally conn.close()
   }
+
+  def getSchemaDefinition(): SchemaDefinition = {
+    val connection = getConnection()
+    try {
+      val metaData = connection.getMetaData
+      val tables = metaData.getTables(null, getSchemaName(metaData), "%", Array("TABLE").map(_.toString))
```
Comment (mproch) [inline engine/components/sql/src/main/scala/pl/touk/nussknacker/sql/db/schema/DbMetaDataProvider.scala:31]: map(_.toString) is redundant?

Comment (mproch) [reply engine/components/sql/src/main/scala/pl/touk/nussknacker/sql/db/schema/DbMetaDataProvider.scala:31]: What about views? I think we also want them (I wonder about synonyms...)

File: engine/components/sql/src/main/scala/pl/touk/nussknacker/sql/service/DatabaseLookupEnricher.scala [hunk 6]
```diff
@@ -42,14 +38,29 @@ object DatabaseLookupEnricher {
 
 class DatabaseLookupEnricher(dBPoolConfig: DBPoolConfig) extends DatabaseQueryEnricher(dBPoolConfig) {
 
+  protected def tableParam(): Parameter = {
+    val schemaMetaData = try {
+      dbMetaDataProvider.getSchemaDefinition()
+    } catch {
+      case e: Exception => SchemaDefinition.empty()
```
Comment (mproch) [inline engine/components/sql/src/main/scala/pl/touk/nussknacker/sql/service/DatabaseLookupEnricher.scala:45]: Let's at least log sth? :) And maybe use scalas: ` case NonFatal(e) => `

File: engine/components/sql/src/main/scala/pl/touk/nussknacker/sql/db/schema/DbMetaDataProvider.scala (hunk 2 above)
Comment (mproch) [reply engine/components/sql/src/main/scala/pl/touk/nussknacker/sql/db/schema/DbMetaDataProvider.scala]: Can we try "connection.getSchema" first? We **have** to use it anyway - or prepend schema to table names...

Comment (dswiecki) [reply engine/components/sql/src/main/scala/pl/touk/nussknacker/sql/db/schema/DbMetaDataProvider.scala]: I'll use connection.getSchema and for various schema we'll instruct the user to use connection string schema setup - thank to that we won'nt need to prepend schema in queries

File: engine/components/sql/src/main/scala/pl/touk/nussknacker/sql/db/schema/DbMetaDataProvider.scala (hunk 5 above)
Comment (dswiecki) [reply engine/components/sql/src/main/scala/pl/touk/nussknacker/sql/db/schema/DbMetaDataProvider.scala]: Added

File: engine/components/sql/src/main/scala/pl/touk/nussknacker/sql/service/DatabaseLookupEnricher.scala (hunk 6 above)
Comment (dswiecki) [reply engine/components/sql/src/main/scala/pl/touk/nussknacker/sql/service/DatabaseLookupEnricher.scala]: Added

**Review by mproch — APPROVED** [review]

**Pull request merged**